ci-operator/step-registry/proxy/config-remove/proxy-config-remove-commands.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ]; then
```

Comment:
Why we need remove these shared files? Ideally any files created in `SHARED_DIR` should not be removed, these shared files are referenced in many places.

Reply:
Thanks for the review. 

During the removal of the management cluster (in `ipi-deprovision-deprovision` for example), if "${SHARED_DIR}/proxy-conf.sh" is found (and sourced), network traffic will be proxied through the non-existent bastion host (which is already removed), causing the step to fail.

Reply:
The bastion host is removed before the hosted cluster (and thus the management cluster) to avoid dependency violations which will otherwise occur during the removal of the hosted cluster. 

This is because the bastion host is located within the hosted cluster's VPC, which cannot be removed without the bastion host being removed first.

Reply:
I have concerns, if the bastion proxy is deleted in advanced, there would be some leftover resources created during the management cluster install.

Reply:
> The bastion host is removed before the hosted cluster (and thus the management cluster) to avoid dependency violations which will otherwise occur during the removal of the hosted cluster.

Personally I feel like this is a bug, per your install workflow, the management cluster is being installed in a user-provisioned vpc as a private cluster, the existing bastion is also user-provisioned, the hosted cluster should not be blocked to destroy because of the pre-existing bastion in the customer vpc.

Reply:
> I have concerns, if the bastion proxy is deleted in advanced, there would be some leftover resources created during the management cluster install.

Thanks for bringing that up. 

The creation of the bastion host is independent of that of the management cluster:
```bash
chain:
  as: cucushift-installer-rehearse-aws-ipi-ovn-hypershift-private-provision
  steps:
  - chain: cucushift-installer-rehearse-aws-ipi-ovn-provision
  ...
  - chain: aws-provision-bastionhost
  ...
```
They correspond to two separate chains respectively. 
I believe that cloud resources will be effectively cleaned up regardless of the order of deletion.

Reply:
> Personally I feel like this is a bug, per your install workflow, the management cluster is being installed in a user-provisioned vpc as a private cluster, the existing bastion is also user-provisioned, the hosted cluster should not be blocked to destroy because of the pre-existing bastion in the customer vpc.

The management cluster is a plain OpenShift IPI cluster on AWS installed via `cucushift-installer-rehearse-aws-ipi-ovn-provision`. The Hypershift hosted cluster is a private one (as opposed to a `Public` or `PublicAndPrivate` one), which implies that the hosted cluster's API server is not exposed to the public.  

The bastion host is provisioned into the (private) hosted cluster's VPC, and blocks the deletion of the latter. This is because a VPC can only be deleted after all cloud resources within are gone.

Reply:
> The management cluster is a plain OpenShift IPI cluster on AWS installed via cucushift-installer-rehearse-aws-ipi-ovn-provision

Ah, my fault, I mis-read that in my last comment, if that, the step makes scenes.